pkg/repo/etcd.go

```diff
@@ -4,32 +4,40 @@ import (
 	"context"
 	"fmt"
 	"log/slog"
+	"sync"
 	"time"
 
 	"go.etcd.io/etcd/clientv3"
 )
 
 type Etcd interface {
-	Put(ctx context.Context, cli *clientv3.Client, key string, value string) error
-	Client(cfg EtcdConfig) (*clientv3.Client, error)
+	Put(key string, value string) error
+
+	Watch(key string) (clientv3.WatchChan, context.CancelFunc)
+
+	Close() error
 }
 
 type EtcdHandler struct {
-	Conf *clientv3.Config
+	sync.RWMutex
```

Comment:
@ksysoev wdyt? is it really a bad practise to embed mutex?

Reply:
If you ask me I'd say struct embedding is bad practice in  general. There are valid use cases  but they are limited.

With lock problem that lock suppose to be internal state of structure, that guarantee safety. when you embed lock all methods of lock became part of public interface